packages/errors/src/test-utils/expect-to-throw-fuel-error.ts

```diff
@@ -0,0 +1,47 @@
+import { isAsyncFunction } from 'util/types';
+
+import type { FuelError } from '../index';
+import { ErrorCode } from '../index';
+
+const enumValues = Object.values(ErrorCode);
+
+function assertExpectations(
+  thrownError: FuelError,
+  expectedError: FuelError | (Partial<FuelError> & Required<Pick<FuelError, 'code'>>)
+) {
+  expect(thrownError.name).toEqual('FuelError');
+
+  expect(enumValues).toContain(expectedError.code);
+
+  (Object.keys(expectedError) as Array<keyof typeof expectedError>).forEach((key) => {
```

Comment:
`expect().toMatchObject()` was giving me some weird errors. This does the same thing and it's straightforward; possibly even more so as the reader doesn't have to understand `toMatchObject`.